### ISSUE TITLE
BUG: nanoseconds argument for DateOffset results in wrong output

### ISSUE THREAD

**Issue by GYHHAHA** [pandas-dev/pandas#36589]
```python
>>>ts = pd.Timestamp('2020-1-1')
>>>ts + pd.DateOffset(nanoseconds=1000)
Timestamp('2020-01-02 00:00:00')
```
This is related to `offset = timedelta(1)` in `_determine_offset` from `pd._libs.tslibs.offsets`.

**Comment by phofl** [comment]
Thanks for your report.

Should we raise in ``_determine_offset`` if the only item in ``kwds`` is ``nanoseconds/s``?

**Comment by GYHHAHA** [comment]
Yes I think so. Users can use nano in Timedelta to handle this instead of DateOffset.

Should I open a PR? I'm not sure if other developers have other opinion.

**Comment by jbrockmendel** [comment]
@tushushu i think this was closed by #43968. can you confirm?

**Comment by tushushu** [comment]
> @tushushu i think this was closed by #43968. can you confirm?

Yes, confirmed with that.
